bot/exts/info/tags.py

```diff
@@ -172,125 +249,168 @@ async def search_tag_content(self, ctx: Context, *, keywords: str) -> None:
+        """  # noqa: D205, D415
+        if tag_name_or_group is None and tag_name is None:
+            if self._tags:
+                await self.list_all_tags(ctx)
                 return True
-            elif founds and len(tag_name) >= 3:
-                await wait_for_deletion(
-                    await ctx.send(
-                        embed=Embed(
-                            title='Did you mean ...',
-                            description='\n'.join(tag['title'] for tag in founds[:10])
-                        )
-                    ),
-                    [ctx.author.id],
-                )
+            else:
+                await ctx.send(embed=Embed(description="**There are no tags!**"))
                 return True
 
-        else:
-            tags = self._cache.values()
-            if not tags:
-                await ctx.send(embed=Embed(
-                    description="**There are no tags in the database!**",
-                    colour=Colour.red()
-                ))
+        if tag_name is None:
+            if any(
+                tag_name_or_group == identifier.group and tag.accessible_by(ctx.author)
+                for identifier, tag in self._tags.items()
+            ):
+                await self.list_tags_in_group(ctx, tag_name_or_group)
                 return True
             else:
-                embed: Embed = Embed(title="**Current tags**")
-                await LinePaginator.paginate(
-                    sorted(
-                        f"**»**   {tag['title']}" for tag in tags
-                        if self.check_accessibility(ctx.author, tag)
-                    ),
-                    ctx,
-                    embed,
-                    footer_text=FOOTER_TEXT,
-                    empty=False,
-                    max_lines=15
-                )
-                return True
-
-        return False
-
-    @tags_group.command(name='get', aliases=('show', 'g'))
-    async def get_command(self, ctx: Context, *, tag_name: TagNameConverter = None) -> bool:
-        """
-        Get a specified tag, or a list of all tags if no tag is specified.
+                tag_name = tag_name_or_group
+                tag_group = None
+        else:
+            tag_group = tag_name_or_group
 
-        Returns True if something can be sent, or if the tag is on cooldown.
-        Returns False if no matches are found.
-        """
-        return await self.display_tag(ctx, tag_name)
+        embed = await self.get_tag_embed(ctx, TagIdentifier(tag_group, tag_name))
+        if embed is not None:
+            if embed is not COOLDOWN.obj:
+                await wait_for_deletion(
+                    await ctx.send(embed=embed),
+                    (ctx.author.id,)
+                )
+            return True
+        else:
+            return False
 
 
 def setup(bot: Bot) -> None:
     """Load the Tags cog."""
     bot.add_cog(Tags(bot))
+
+
+def extract_tag_identifier(string: str) -> TagIdentifier:
```

Comment:
How does moving this to a classmethod sound?

Reply:
Sounds good, made it a constructor of the class in 359a17c00bb036f9a90fe7681592dee29f84c806